docs/content/sips/018-application-metadata.md

```diff
@@ -0,0 +1,73 @@
+title = "SIP 018 - Application Metadata"
+template = "main"
+date = "2024-04-12T23:00:00Z"
+
+---
+
+Summary: Allow guest code to access information from the application manifest
+
+Owner(s): ivan.towlson@fermyon.com
+
+Created: Apr 12, 2024
+
+# Requirements and scope
+
+We have, over time, had a few requests for guest code to be able to read fields from the application manifest, or from the host itself. Examples include:
+
+| Field                                         | Purpose          | Notes |
+|-----------------------------------------------|------------------|-------|
+| Application name and version                  | Logging          |       |
+| Host runtime name or other identifying string | Logging          |       |
+| Host runtime version                          | Logging          |       |
+| Component id                                  | Logging (?)      |       |
+| Files mounted to the component                | Not given        | Already available by traversing the component's file system |
+| Allowed outbound hosts for the component      | Not given        | One user mentioned their app needed to scan a set of sites but I don't have details |
```

Comment:
This is one that I am interested in also.
A use case would be a Spin application that accepts jobs to check for broken links. The Spin application fetches links (broken link check jobs) to be checked from a central database. To prevent asking for jobs that the application is not willing to check, the application could send a list of links it is prepared to check when asking for jobs. But to do this the application needs to have access to the `allowed_outbound_hosts` config i.e. `["https://www.fermyon.com/", "https://developer.fermyon.com"]. The central database can filter on these hosts and just send actionable links back to the Spin app.

Reply:
This would save double handling a list of hosts in the config and a list of hosts that the application can access while running.

Reply:
Here is an example of the workflow between a Spin application that checks links and the central database.

![image](https://github.com/fermyon/spin/assets/9831342/88214705-5e08-41e5-8648-a757364620e3)

Reply:
FWIW, I would generally expect a link checker to have `allowed_outbound_hosts = ["https://*"]`.

Reply:
That is a fair point :heavy_check_mark: